libarchive/archive_read_support_format_tar.c

```diff
@@ -2546,20 +2559,16 @@ pax_attribute(struct archive_read *a, struct tar *tar, struct archive_entry *ent
 			else if (key_length > 6 && memcmp(key, "xattr.", 6) == 0) {
 				key_length -= 6;
 				key += 6;
-				if (value_length > xattr_limit) {
```

Comment:
This is the right place to check the limit.  If the limit here is too small, doubling it would be entirely reasonable.

We should have a better error message here, which includes the name of the xattr.